src/bam/record.rs

```diff
@@ -81,6 +81,7 @@ impl PartialEq for Record {
             && self.mpos() == other.mpos()
             && self.insert_size() == other.insert_size()
             && self.data() == other.data()
+            && self.inner().core.l_extranul == other.inner().core.l_extranul
```

Comment:
Extranul does not really change anything in terms of semantics, right? I wonder if we rather should consider two records the same regardless of the amount of padding.

Reply:
If I were to recommend any changes to the API, qname_len() should probably be modified to take l_extranul into account (it would make more sense--someone is probably more interested in the (&str-like) bytes of the header instead of the &CStr + [0-3] bytes of (extra) null-padding.

Reply:
Yes, I agree.

Reply:
If you like, please do that in this PR.

Reply:
Excellent. Now I think it would make sense to also remove the l_extranul check above, wouldn't it?

Reply:
I don't think so--the l_extranul comparison is still required to guarantee that qname() yields the same output between objects otherwise identical. All other non-variable data are either implicitly captured by the data() comparison or are explicitly represented--all except l_extranul.

Reply:
I can remove it (I don't think it looks very pleasant either), but then e.g. test_bincode() and test_serde_json() may pass, even if l_extranul handling is broken.

Alternatively, I could implement Eq for bam1_core_t instead?

Something else I now notice, looking more closely--I think we should take care to align the beginning of the cigar as intended (the whole purpose behind l_extranul). If you agre, feel free to make an issue and assign it to me.

Reply:
Yes, implementing Eq for bam1_core_t makes more sense indeed.